test/lit.cfg

```diff
@@ -1339,36 +1338,22 @@ elif run_os == 'linux-androideabi' or run_os == 'linux-android':
     config.target_runtime = "native"
     config.target_swift_autolink_extract = inferSwiftBinary("swift-autolink-extract")
     config.target_sdk_name = "android"
-    android_link_paths_opt = "-L {} -L {} -L {}".format(
-        shell_quote(make_path(
-            config.android_ndk_path, "sources", "cxx-stl", "llvm-libc++",
-            "libs", ndk_platform_tuple)),
+    android_link_paths_opt = "-L {} -L {}".format(
         shell_quote(make_path(
             toolchain_directory, "lib", "gcc", ndk_platform_triple,
             "{}.x".format(config.android_ndk_gcc_version))),
         shell_quote(make_path(
             toolchain_directory, ndk_platform_triple, "lib")))
-    # Since NDK r14 the headers are unified under $NDK_PATH/sysroot, so the -sdk
-    # switch is not enough. Additionally we have to include both the unified
-    # sysroot, and the architecture sysroot.
-    unified_android_include_path = shell_quote(make_path(
-        config.android_ndk_path, "sysroot", "usr", "include"))
-    architecture_android_include_path = shell_quote(make_path(
-        config.android_ndk_path, "sysroot", "usr", "include",
-        ndk_platform_triple))
-    android_include_paths_opt = "-I {} -I {}".format(
-        unified_android_include_path, architecture_android_include_path)
-    # clang can use -isystem, but Swift cannot.
-    android_include_system_paths_opt = "-isystem {} -isystem {}".format(
-        unified_android_include_path, architecture_android_include_path)
+    resource_dir_opt = ("-resource-dir %s" % test_resource_dir)
```

Comment:
I just added this single line, should fix the testing issues.

Reply:
There's still something missing.

The following tests seem to fail: https://gist.github.com/drodriguez/812f2f0063ede38e87ff830f966c344a

I was checking the first one with:

```
$ env ANDROID_SKIP_CLEAN=1 swift/utils/run-test --build-dir <build_dir>/swift-linux-x86_64/ --target android-aarch64 --mode only_non_executable swift/test/AutoDiff/compiler_crashers_fixed/sr13933-vjpcloner-apply-multiple-consuming-users.swift -v
```

The failing command is:

```
[...]/swift-source/build/my_android_aarch64/swift-linux-x86_64/bin/swiftc -target aarch64-unknown-linux-android -sdk [...]/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/sysroot -tools-directory [...]/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/aarch64-linux-android/bin -L [...]/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/lib/gcc/aarch64-linux-android/4.9.x -L [...]/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/aarch64-linux-android/lib -use-ld=gold -resource-dir <build_dir>/swift-linux-x86_64/lib/swift -module-cache-path <build_dir>/swift-linux-x86_64/swift-test-results/aarch64-unknown-linux-android/clang-module-cache -swift-version 4  -Xfrontend -ignore-module-source-info  [...]/swift-source/swift/test/AutoDiff/compiler_crashers_fixed/sr13933-vjpcloner-apply-multiple-consuming-users.swift
```

The error is:

```
[...]/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/aarch64-linux-android/bin/ld.gold: error: cannot open crtbegin_dynamic.o: No such file or directory
[...]/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/aarch64-linux-android/bin/ld.gold: error: cannot open crtend_android.o: No such file or directory
bionic/libc/bionic/libc_init_common.cpp:117: error: undefined reference to 'pthread_atfork'
external/jemalloc/src/jemalloc.c:1422: error: undefined reference to 'pthread_atfork'
external/jemalloc/src/jemalloc.c:1422: error: undefined reference to 'pthread_atfork'
external/jemalloc/src/jemalloc.c:1422: error: undefined reference to 'pthread_atfork'
external/jemalloc/src/jemalloc.c:1358: error: undefined reference to 'atexit'
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
<unknown>:0: error: link command failed with exit code 1 (use -v to see invocation)
```

The problem is that `crtbegin_dynamic.o` and `crtend_android.o` are not actually found in any of the directories that the compiler driver is looking at (for reference, in my copy they are in `android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/aarch64-linux-android/21/` and other similar directories, and in the not unified path, it was in `android-ndk-r19c/platforms/android-21/arch-arm64/usr/lib`).

Passing that directory as a `-L` doesn't seem to do the trick. Passing as a `-tools-directory` seems to find the right one (because it gets turn into `-B`), but Swift only seems to allow one `-tools-directory`, and one loses access to the linker.

Reply:
It's the Android API missing from the target triple at the end, clang needs that to figure out the path to the right libraries in the unified sysroot.

Reply:
I think I tried modifying manually the target and adding the API level, but it didn't seem to make any difference. It should be possible to pass the actual value through the `lit.site.cfg.in`, but I didn't try that change (since it was not working when changing it manually).

Reply:
I'll take a look and see if I can reproduce.